app/src/features/surveys/telemetry/ManualTelemetryList.tsx

```diff
@@ -0,0 +1,562 @@
+    }
+  };
+
+  const handleSubmit = async (data: AnimalDeployment) => {
+    if (formMode === ANIMAL_FORM_MODE.ADD) {
+      // ADD NEW DEPLOYMENT
+      await handleAddDeployment(data);
+    } else {
+      // EDIT EXISTING DEPLOYMENT
+      await handleEditDeployment(data);
+    }
+    // UPLOAD/ REPLACE ANY FILES FOUND
+    if (data.attachmentFile && data.attachmentType) {
+      await handleUploadFile(data.attachmentFile, data.attachmentType);
+    }
+  };
+
+  const handleDeleteDeployment = async () => {
+    try {
+      const deployment = deployments?.find((item) => item.device_id === deviceId);
+      if (!deployment) {
+        throw new Error('Invalid Deployment Data');
+      }
+      const critter = critters?.find((item) => item.critter_id === deployment?.critter_id);
+      if (!critter) {
+        throw new Error('Invalid Critter Data');
+      }
+
+      const found = telemetryContext.telemetryDataLoader.data?.find(
+        (item) => item.deployment_id === deployment.deployment_id
+      );
+      if (!found) {
+        await biohubApi.survey.removeDeployment(
+          surveyContext.projectId,
+          surveyContext.surveyId,
+          critter.survey_critter_id,
+          deployment.deployment_id
+        );
+        dialogContext.setYesNoDialog({ open: false });
+        surveyContext.deploymentDataLoader.refresh(surveyContext.projectId, surveyContext.surveyId);
+      } else {
+        dialogContext.setYesNoDialog({ open: false });
+        // Deployment is used in telemetry, do not delete until it is scrubbed
+        throw new Error('Deployment is used in telemetry');
+      }
+    } catch (e) {
+      dialogContext.setSnackbar({
+        snackbarMessage: (
+          <>
+            <Typography variant="body2" component="div">
+              <strong>Error Deleting Deployment</strong>
+            </Typography>
+            <Typography variant="body2" component="div">
+              {String(e)}
+            </Typography>
+          </>
+        ),
+        open: true
+      });
+    }
+  };
+
+  const deleteDeploymentDialog = () => {
+    dialogContext.setYesNoDialog({
+      dialogTitle: 'Delete Deployment?',
+      dialogContent: (
+        <Typography variant="body1" component="div" color="textSecondary">
+          Are you sure you want to delete this deployment?
+        </Typography>
+      ),
+      yesButtonLabel: 'Delete Deployment',
+      noButtonLabel: 'Cancel',
+      yesButtonProps: { color: 'error' },
+      onClose: () => {
+        dialogContext.setYesNoDialog({ open: false });
+      },
+      onNo: () => {
+        dialogContext.setYesNoDialog({ open: false });
+      },
+      open: true,
+      onYes: () => {
+        handleDeleteDeployment();
+      }
+    });
+  };
+
+  const handleAddDeployment = async (data: AnimalDeployment) => {
+    const critter = critters?.find((a) => a.survey_critter_id === data.survey_critter_id);
+    if (!critter) console.log('Did not find critter in addTelemetry!');
```

Comment:
Is this needed? Presumably the following code can handle the case where the critter is null?